### DISCUSSION TITLE
How to use KEDA with Knative serving (not eventing)?

### DISCUSSION THREAD

**Discussion by nicoga-code** [kedacore/keda#4462] [Q&A / Need Help]
Hi, in the company I work for we are looking for tools in order to be able to auto scale pods based on kafka lag, it seems KEDA is a perfect fit for this the problem is that our infrastructure is using Knative serving, not eventing, it is possible to use KEDA with Knative serving and if yes how can we do it?

Thanks

**Comment by zroubalik** [comment]
@nicoga97 could you please make a digest from our conversation here?

**Comment by nicoga-code** [comment]
Sure  ! So we need to use KEDA with knative serving  because changing to knative eventing would be a huge change because how we have configured our infrastructure so the idea is to start using HPA instead of KPA and create a new controller that will sync between Knative Service and ScaledObject using the exposed subresource `/scale' do I missed anything @zroubalik?
